### DISCUSSION TITLE
Experience/Example authenticating single processes with polkit

### DISCUSSION THREAD

**Discussion by lukeflo** [ratatui/ratatui#1599] [Q&A]
I write a little personal TUI to try out some things and learn Rust. For this project I need to run single external commands (spawned via `std::process::command`) with root permissions while the app itself should run without these.

The preferred auth method for such tasks on Linux seems to be using `polkit`. Unfortunately, there are very few informational resources out there which explain how to set this up. And the specific crates, [`zbus_polkit`](https://crates.io/crates/zbus_polkit) and [`polkit`](https://crates.io/crates/polkit), do not offer detailed docs; if any.

Therefore, I decided to ask here at the correct TUI-specific place, if anybody knows some good resources with code examples or a crate/project which can serve as template for such a polkit-based authentication task.

I'm happy for every hint!

**Comment by joshka** [comment]
polkit is written in C. The rust crates are just abstractions on top of the C libs. To understand polkit, you're likely going to have to read the docs in the original form, and then translate the concepts you've learnt over to Rust. When doing this sort of thing, expect that there is always going to be some things which are lost in translation which will make it difficult as well as bugs. https://polkit.pages.freedesktop.org/polkit/  and https://github.com/polkit-org/polkit?tab=readme-ov-file are your start points.

There's nothing really Ratatui specific about this. Good luck :)

**Reply by lukeflo** [reply]
Yeah, thanks for the tipps. I don't know much C, so this would be a challenge. And the Rust crates providing an API to `polkit` are not documented too detailed.

I was hoping that someone already has tried something similar for a ratatui-based app which could serve as template. But maybe its too much overhead. There are other ways to authenticate for a single process...

Happy new year!

**Reply by joshka** [reply]
Your best bet for finding that sort of thing might be to use github search to help.
https://github.com/search?q=path%3ACargo.toml%20polkit&type=code
